### PR TITLE
Azure Database Configuration

### PR DESCRIPTION
**Pull request by ytsarev**
### Description of your changes

Full config, examples, uptest

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

local `make e2e` + uptest below

### CODE REVIEW

**Comment by ytsarev** [comment]
ok, forgot to setup creds :D

File: apis/definition.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                    description: "A reference to the Secret object containing database password"
+                    properties:
+                      namespace:
+                        type: string
+                      name:
+                        type: string
+                      key:
+                        type: string
+                    required:
+                    - namespace
```
Comment (haarchri) [inline apis/definition.yaml:59]: Intendation

File: apis/definition.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  claimNames:
+    kind: SQLInstance
+    plural: sqlinstances
+  connectionSecretKeys:
+    - username
+    - password
+    - port
+    - host
+  versions:
+  - name: v1alpha1
```
Comment (haarchri) [inline apis/definition.yaml:19]: Intendation

File: crossplane.yaml [hunk 3]
```diff
@@ -11,9 +11,11 @@ spec:
   crossplane:
     version: ">=v1.13.2-0"
   dependsOn:
+    - provider: xpkg.upbound.io/upbound/provider-family-azure
```
Comment (haarchri) [inline crossplane.yaml:14]: why we need this explicit ?

File: apis/definition.yaml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                    - key
+                  networkRef:
+                    type: object
+                    description: "A reference to the XNetwork object that this Database should be connected to."
+                    properties:
+                      id:
+                        type: string
+                        description: ID of the Network object this ref points to.
+                    required:
+                    - id
```
Comment (haarchri) [inline apis/definition.yaml:70]: Intendation

File: crossplane.yaml (hunk 3 above)
Comment (ytsarev) [reply crossplane.yaml:14]: Because it installs the latest/`v0.38.1` as of now and eventually conflicts :)

Comment (haarchri) [reply crossplane.yaml:14]: but this is a bug in crossplane package manager or ? why the provider dep is installing `v0.38.1` ?

Comment (ytsarev) [reply crossplane.yaml:14]: @haarchri this is known behavior of the package manager, I will dig up the discussion

Comment (ytsarev) [reply crossplane.yaml:14]: @haarchri please see https://github.com/upbound/docs/pull/236/files

**Review by haarchri — APPROVED** [review]
Thanks

**Pull request merged**